tests/test_e2e/test_secrets.py

```diff
@@ -36,97 +36,91 @@ def test_get_secret_no_secret(mycharm):
             assert mgr.charm.model.get_secret(label="foo")
 
 
-def test_get_secret(mycharm):
+@pytest.mark.parametrize("owner", ("app", "unit"))
+def test_get_secret(mycharm, owner):
     ctx = Context(mycharm, meta={"name": "local"})
+    secret = Secret({"a": "b"}, owner=owner)
     with ctx.manager(
-        state=State(secrets={Secret(id="foo", contents={0: {"a": "b"}})}),
+        state=State(secrets={secret}),
         event=ctx.on.update_status(),
     ) as mgr:
-        assert mgr.charm.model.get_secret(id="foo").get_content()["a"] == "b"
+        assert mgr.charm.model.get_secret(id=secret.id).get_content()["a"] == "b"
 
 
 @pytest.mark.parametrize("owner", ("app", "unit"))
 def test_get_secret_get_refresh(mycharm, owner):
     ctx = Context(mycharm, meta={"name": "local"})
+    secret = Secret(
+        tracked_content={"a": "b"},
+        latest_content={"a": "c"},
+        owner=owner,
+    )
     with ctx.manager(
         ctx.on.update_status(),
-        State(
-            secrets={
-                Secret(
-                    id="foo",
-                    contents={
-                        0: {"a": "b"},
-                        1: {"a": "c"},
-                    },
-                    owner=owner,
-                )
-            }
-        ),
+        State(secrets={secret}),
     ) as mgr:
         charm = mgr.charm
-        assert charm.model.get_secret(id="foo").get_content(refresh=True)["a"] == "c"
+        assert (
+            charm.model.get_secret(id=secret.id).get_content(refresh=True)["a"] == "c"
+        )
 
 
 @pytest.mark.parametrize("app", (True, False))
 def test_get_secret_nonowner_peek_update(mycharm, app):
     ctx = Context(mycharm, meta={"name": "local"})
+    secret = Secret(
+        {"a": "b"},
+        {"a": "c"},
```

Comment:
Seeing this in "real life" makes me regret my earlier "don't have strong feelings" view on this: it's not at all clear which is the latest and which is tracked. I really don't think we should allow two of the same type as position. Compare to the example above on lines 53-57, where it's ultra-obvious. Can we make `latest_content` keyword-only after all?

Reply:
Discussed this in person briefly also. I'm on board with starting out restrictive and then lifting that if later on we decide it wasn't the right choice (much easier than the other way around). I do still feel like old,new is the intuitive ordering.